File: c++/data_src_dependent/grid_from_bridges/grid_from_bridges.cpp
#include <string>
#include "Bridges.h"
#include "DataSource.h"
#include "ColorGrid.h"


using namespace std;
using namespace bridges;

int main(int argc, char **argv) {
#if TESTING
	// command line args provide credentials and server to test on
	Bridges *bridges =  new Bridges(atoi(argv[1]), argv[2], argv[3]);
	if (argc > 4)
		bridges->setServer(argv[4]);
#else
	Bridges *bridges =  new Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID",
		"YOUR_API_KEY");
#endif

	//First make a ColorGrid
	int sizeI = 25, sizeJ = 35;
	ColorGrid cg1 (sizeI, sizeJ);
	for (int i = 0; i < sizeI; ++i)
		for (int j = 0; j < sizeJ; ++j) {
			if ((i + j) % 2 == 0) {
				cg1.set(i, j, Color(0, 0, 0));
			}
			else {
				cg1.set(i, j, Color(255, 255, 255));
			}
		}
	bridges->setDataStructure(&cg1);
	bridges->visualize();

	try {
		DataSource *ds = new DataSource;
		ColorGrid cg = ds->getColorGridFromAssignment(bridges->getUserName(), bridges->getAssignment());

		// just to be sure this is working, lets change the mid square
		// of pixels to a red colored square

		int *dims = cg.getDimensions();
		cout << "Dims:" << dims[0] << "," << dims[1] << endl;
		for (int k = dims[1] / 2 - 5; k < dims[1] / 2 + 5; k++)
			for (int j = dims[0] / 2 - 5; j < dims[0] / 2 + 5; j++)
				cg.set(k, j, Color("red"));
		bridges->setDataStructure(&cg);
		bridges->visualize();

		delete ds;
	}
	catch (char const * s) {
		std::cerr << "exception caught: " << s << std::endl;
		return -1;
	}

	delete bridges;

	return 0;
}

File: python/data_src_dependent/grid_from_bridges/grid_from_bridges.py
from bridges.bridges import *
from bridges.color_grid import *
from bridges.color import *
import sys

def main():
    args = sys.argv[1:]

    # create the Bridges object, set credentials

    bridges = Bridges(int(args[0]), args[1], args[2])

    if len(args) > 3:
        bridges.connector.set_server(args[3])

    sizei = 25
    sizej = 35

    cg1 = ColorGrid(sizei, sizej)

    for i in range(sizei):
        for j in range(sizej):
            if (i+j) % 2 == 0:
                cg1.set(i,j,Color(0,0,0))
            else:
                cg1.set(i,j,Color(255,255,255))

    bridges.set_data_structure(cg1)
    bridges.visualize()

    cg = bridges.get_color_grid_from_assignment(bridges.get_username(), bridges.get_assignment_id(), 0)

    dims = cg.get_dimensions()
    for k in range(int(dims[1] / 2 - 5), int(dims[1]/2+5)):
        for j in range(int(dims[0]/2-5), int(dims[0]/2 + 5)):
            cg.set(k,j,Color("red"))

    bridges.set_data_structure(cg)
    bridges.visualize()

if __name__ == '__main__':
    main()

